### ISSUE TITLE
flaky TestBackoffConnector test

### ISSUE THREAD

**Issue by marten-seemann** [libp2p/go-libp2p#1313]
```
=== RUN   TestBackoffConnector
      backoffconnector_test.go:107: wrong number of connections. expected 3, actual 4
  --- FAIL: TestBackoffConnector (1.89s)

**Comment by marten-seemann** [comment]
```
  === RUN   TestBackoffConnector
      backoffconnector_test.go:85: 
          	Error Trace:	backoffconnector_test.go:85
          	Error:      	Condition never satisfied
          	Test:       	TestBackoffConnector
  --- FAIL: TestBackoffConnector (3.07s)

**Comment by marten-seemann** [comment]
Fixed by #1316.
